File: src/main/java/com/demonwav/statcraft/magic/BucketCode.java
/*
 * StatCraft Bukkit Plugin
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 * http://demonwav.com
 *
 * MIT License
 */

package com.demonwav.statcraft.magic;

public enum BucketCode {

    WATER((byte)0),
    LAVA((byte)1),
    MILK((byte)2);

    private byte code;

    BucketCode(byte code) {
        this.code = code;
    }

    public byte getCode() {
        return code;
    }

    public static BucketCode fromCode(byte code) {
        for (BucketCode bucketCode : values()) {
            if (code == bucketCode.getCode())
                return bucketCode;
        }
        return null;
    }
}

File: src/main/kotlin/com/demonwav/statcraft/magic/BucketCode.kt
/*
 * StatCraft Bukkit Plugin
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 * https://www.demonwav.com
 *
 * MIT License
 */

package com.demonwav.statcraft.magic

enum class BucketCode(val code: Byte) {

    WATER   (0.toByte()),
    LAVA    (1.toByte()),
    MILK    (2.toByte());

    companion object {
        @JvmStatic
        fun fromCode(code: Byte): BucketCode? {
            for (bucketCode in values()) {
                if (code == bucketCode.code)
                    return bucketCode
            }
            return null
        }
    }
}
